client/main.lua

```diff
@@ -66,33 +67,38 @@ local function getVehicle()
     return vehicle
 end
 
----manages the opening of locks
+---Manages the opening of locks
 ---@param vehicle number? The entity number of the vehicle.
 ---@param state boolean? State of the vehicle lock.
----@param anim any Aniation
+---@param anim any Animation
 local function setVehicleDoorLock(vehicle, state, anim)
     if not vehicle then return end
     if not isBlacklistedVehicle(vehicle) then
         if hasKeys(qbx.getVehiclePlate(vehicle)) or areKeysJobShared(vehicle) then
-
             if anim then
                 lib.requestAnimDict('anim@mp_player_intmenu@key_fob@')
                 TaskPlayAnim(cache.ped, 'anim@mp_player_intmenu@key_fob@', 'fob_click', 3.0, 3.0, -1, 49, 0, false, false, false)
             end
-
             TriggerServerEvent('InteractSound_SV:PlayWithinDistance', 5, 'lock', 0.3)
             NetworkRequestControlOfEntity(vehicle)
 
             local lockstate
             if state ~= nil then
                 lockstate = state and 2 or 1
             else
-                lockstate = (GetVehicleDoorLockStatus(vehicle) % 2) + 1 -- (1 % 2) + 1 -> 2  (2 % 2) + 1 -> 1 
+                local currentLockState = GetVehicleDoorLockStatus(vehicle)
+                local isPedInVehicle = cache.vehicle == vehicle
+
+                if childLockEnabled and isPedInVehicle and currentLockState == 4 then
+                    -- Check if the player is in the vehicle and the lock state is 4, then switch it to 1
+                    lockstate = 1
+                else
+                    lockstate = isPedInVehicle and 4 or (currentLockState % 2) + 1 -- Use state 4 if the player is inside, otherwise 1 or 2
```

Comment:
Will someone be able to get in the car while someone is sitting in?

Reply:
What if the locks are open and someone tries to sit as a passenger?

Reply:
Of course, when the driver is already in the car.

Reply:
mmmh thank you for making me think that I will adjust the code, I will adjust the code